Source/SpatialGDK/Private/SpatialInterop.cpp

```diff
@@ -860,16 +867,148 @@ bool USpatialInterop::IsSingletonClass(UClass* Class)
 	return false;
 }
 
-NameToEntityIdMap* USpatialInterop::GetSingletonNameToEntityId() const
+improbable::unreal::GlobalStateManagerData* USpatialInterop::GetGlobalStateManagerData() const
 {
 	TSharedPtr<worker::View> View = SpatialOSInstance->GetView().Pin();
 	auto It = View->Entities.find(SpatialConstants::GLOBAL_STATE_MANAGER);
 
 	if (It != View->Entities.end())
 	{
 		improbable::unreal::GlobalStateManagerData* GSM = It->second.Get<improbable::unreal::GlobalStateManager>().data();
-		return &(GSM->singleton_name_to_entity_id());
+		return GSM;
 	}
 
 	return nullptr;
 }
+
+NameToEntityIdMap* USpatialInterop::GetSingletonNameToEntityId() const
+{
+	improbable::unreal::GlobalStateManagerData* GSM = GetGlobalStateManagerData();
+
+	if (GSM == nullptr)
+	{
+		return nullptr;
+	}
+
+	return &(GSM->singleton_name_to_entity_id());
+}
+
+EntityIdToPathMap* USpatialInterop::GetEntityIdToReplicatedStablyNamedPath() const
+{
+	improbable::unreal::GlobalStateManagerData* GSM = GetGlobalStateManagerData();
+
+	if (GSM == nullptr)
+	{
+		return nullptr;
+	}
+
+	return &(GSM->entity_id_to_replicated_stably_named_path());
+}
+
+void USpatialInterop::ReserveReplicatedStablyNamedActorChannel(USpatialActorChannel* Channel)
+{
+	ReplicatedStablyNamedActorChannelQueue.Add(Channel);
+
+	if (CanSpawnReplicatedStablyNamedActors())
+	{
+		ReserveReplicatedStablyNamedActors();
+	}
+}
+
+void USpatialInterop::UnreserveReplicatedStablyNamedActor(AActor* Actor)
+{
+	// Slightly inefficient linear search, could use a hashmap
+	USpatialActorChannel* Channel = nullptr;
+	for (auto& It : ReplicatedStablyNamedActorChannelQueue)
+	{
+		if (It->Actor == Actor)
+		{
+			Channel = It;
+			break;
+		}
+	}
+
+	if (Channel != nullptr)
+	{
+		ReplicatedStablyNamedActorChannelQueue.Remove(Channel);
+	}
+}
+
+void USpatialInterop::AddReplicatedStablyNamedActorToGSM(const FEntityId& EntityId, AActor* Actor)
+{
+	EntityIdToPathMap& EntityIdToReplicatedStablyNamedPathMap = *GetEntityIdToReplicatedStablyNamedPath();
+	// If the map already has the entity id, meaning the actor was already spawned, return early
+	if (EntityIdToReplicatedStablyNamedPathMap.count(EntityId.ToSpatialEntityId()) != 0)
+	{
+		return;
+	}
+
+	std::string Path = TCHAR_TO_UTF8(*Actor->GetPathName(Actor->GetWorld()));
+	EntityIdToReplicatedStablyNamedPathMap.emplace(EntityId.ToSpatialEntityId(), Path);
+
+	improbable::unreal::GlobalStateManager::Update Update;
+	Update.set_entity_id_to_replicated_stably_named_path(EntityIdToReplicatedStablyNamedPathMap);
+
+	TSharedPtr<worker::Connection> Connection = SpatialOSInstance->GetConnection().Pin();
+	Connection->SendComponentUpdate<improbable::unreal::GlobalStateManager>(worker::EntityId((long)SpatialConstants::GLOBAL_STATE_MANAGER), Update);
+}
+
+void USpatialInterop::ReserveReplicatedStablyNamedActors()
```

Comment:
What about RegisterReplicatedStablyNamedActors? Better implies that this is an authoritative server action.